internal/appsec/emitter/waf/addresses/builder.go

```diff
@@ -47,7 +47,7 @@ func (b *RunAddressDataBuilder) WithHeadersNoCookies(headers map[string][]string
 
 func (b *RunAddressDataBuilder) WithCookies(cookies map[string][]string) *RunAddressDataBuilder {
 	if len(cookies) == 0 {
-		cookies = nil
+		return b
 	}
```

Comment:
This means `WithCookies(notNil)` followed by `WithCookies(nil)` will leave the old cookies value around... 

That feels like a pretty theoretical issue... but I guess if this is the intended behavior it should probably be documented at least?

Reply:
Actually this is the behavior for a lot of other optional addresses managed by the address builder. I had to temporarily change it otherwise fingerprints would not generate when there was no cookies in the request but the original goal is still the same: if the map is empty then there nothing for the WAF to run on, so we don't keep nil values. We could also try to delete the previous key in this case but this looks a bit overengineered considering the builder is a short-lived value.

Reply:
I don't mind adding docs on the WafAddressesBuilder type tho